### DISCUSSION TITLE
State dependency question

### DISCUSSION THREAD

**Discussion by junjielu** [pointfreeco/swift-composable-architecture#1029] [Q&A]
I've recently been using TCA to build my app, and I think TCA is very powerful for expressing the structure and state of the app. However, I've run into problems expressing state dependencies within the app through the nested form of struct.

One typical example is deep link, the design of deep link breaks this nested expression. For example, `nav://app/foo` may instruct `UINavigationController` to push a specific `UIViewController`, which in the design of State requires adding the dependency of this state to the `AppState`. 

The question is if the majority of UIViewControllers in the app are deep-link capable, will all the dependencies be tiled in the AppState, and if so, how will the state in the UINavigationController stack be expressed? Looking forward to someone answering my question!

**Comment by JacksonUtsch** [comment]
Does the "dependency" state require persistence? I don't think it should generally. Then it could be represented in a enum's associated type.

`nav://app/foo(FooState)`

Makes accessing the state a bit trickier but more of an accurate model.
